src/pages/profile.tsx

```diff
@@ -78,7 +108,88 @@ export default function Page() {
       <Input background="white" 
         value={newCity} onChange={e => setNewCity(e.target.value)} />
     </FormControl>           
+
+    <Divider margin={componentSpacing} />
+
+    <InterviewPreference data={unsaved?.interviews} 
+      updateData={updateInterviewPref} /> 
+
     <Button onClick={handleSubmit} variant="brand">保存</Button>
     {isLoading && <Loader loadingText='保存中...'/>}
   </VStack>;
 }
+
+function InterviewPreference({ data, updateData } : {
+  data: InterviewPref;
+  updateData: (data: InterviewPref) => void;
+}) {
+  const until = data?.limit?.until ? new Date(data.limit.until) : oneMonthDate;
+  const noMoreThan = data?.limit?.noMoreThan || 0;
+
+  const setLimit = (noMoreThan: number, until: Date) => {
+    if (noMoreThan !== undefined && until !== undefined) {
+      const newData = structuredClone(data || {});
+      newData.limit = { noMoreThan, until: until.toISOString() };
```

Comment:
can you similar to line 146, use the `{...data, ...}` approache?

same question for removeLimit and updateInterviewPref